osf/management/commands/force_archive.py

```diff
@@ -280,48 +281,72 @@ def get_logs_to_revert(reg):
             Q(node=reg.registered_from) |
             (Q(params__source__nid=reg.registered_from._id) | Q(params__destination__nid=reg.registered_from._id))).order_by('-date')
 
+def get_file_obj_from_log(log, reg):
+    try:
+        return BaseFileNode.objects.get(_id=log.params['urls']['view'].split('/')[4])
+    except KeyError:
+        if log.action == 'osf_storage_folder_created':
+            return OsfStorageFolder.objects.get(
+                target_object_id=reg.registered_from.id,
+                name=log.params['path'].split('/')[-2]
+            )
+        elif log.action == 'osf_storage_file_removed':
+            path = log.params['path'].split('/')
+            return TrashedFileNode.objects.get(
+                target_object_id=reg.registered_from.id,
+                name=path[-1] or path[-2]  # file name or folder name
+            )
+        elif log.action in ['addon_file_moved', 'addon_file_renamed']:
+            try:
+                return BaseFileNode.objects.get(_id=log.params['source']['path'].rstrip('/').split('/')[-1])
+            except (KeyError, BaseFileNode.DoesNotExist):
+                return BaseFileNode.objects.get(_id=log.params['destination']['path'].rstrip('/').split('/')[-1])
+        else:
+            # Generic fallback
+            path = log.params.get('path', '').split('/')
+            if len(path) >= 2:
+                name = path[-1] or path[-2]  # file name or folder name
+                return BaseFileNode.objects.get(
+                    target_object_id=reg.registered_from.id,
+                    name=name
+                )
```

Comment:
Minor: Same possible uniqueness issue, but I'm not sure what cases this generic fallback would catch. Could be improved by filtering for `content_type_id`, to at least mitigate naming collisions across object types, which is a greater risk for files than folders.